diesel/src/main.rs

```diff
@@ -56,6 +59,73 @@ fn index(
         .responder()
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct MyUser {
+    name: String
+}
+
+const MAX_SIZE: usize = 262_144; // max payload size is 256k
+
+/// This handler manually load request payload and parse json object
+fn index_add(req: &HttpRequest<AppState>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    // HttpRequest::payload() is stream of Bytes objects
+    req.payload()
```

Comment:
You probably want here not request, but state extractor https://actix.rs/actix-web/actix_web/struct.State.html

Reply:
There seems to be no way to mix a config and state extractor.

I also want to say, I really appreciate your feedback here, and I know you are trying to "advise" to a simpler way, but there a good reasons that I am asking for the examples I am. I want to be able to respond with complex error messages about failures in the parsing of the body (extractors block this), I want to eventually be able to decode types other than json (CBOR etc). I want to have DoS protections (message size limits and more). I also want to be able to standardise on a single json parser (serde). 

So I really appreciate your advice here, but I would really like your help to fix the examples I have here: both the manual parsing example and the extractor async example. I'm sure that other people will run into these problems in the future, and so getting good, advanced use cases that are production ready as examples will really help adoption of the framework,

Thanks,